### PR TITLE
travis: run tests in 2 processes

### PR DESCRIPTION
**Pull request by MetRonnie**
Cherry picked all the commits from #164 but changed it to 2 processes (as 4 doesn't run any faster on the dual-core VM) and fixed the Travis timeout.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -38,32 +38,26 @@ install:
...
   - pip install --upgrade pip setuptools
-  # Coverage dependencies, coverage for reporting coverage, pytest-cov to
-  # enable PyTest to produce coverage reports, and pytest-env to allow
-  # PyTest to use environment variables in its configuration (e.g. TZ=UTC).
-  - pip install coverage pytest-cov pytest-env pytest-runner
-  # Code style dependencies
-  - pip install pycodestyle
-  - pip install -e .
+  - pip install -e .[all]
+  - pip install coverage pytest-xdist pytest-cov pycodestyle
```
Comment (kinow) [inline .travis.yml:42]: I think this line is not necessary? In `setup.py` you are adding `tests_require` to `all` dependencies subgroup already?

**Review by kinow — APPROVED** [review]

**Comment by oliver-sanders** [comment]
>  but changed it to 2 processes (as 4 doesn't run any faster on the dual-core VM)

Fair enough I guess. I tend to overload processes because of context swapping, hyperthreading, IO etc. When it comes to virtual hardware it's somewhat harder to predict what the impact would be.

**Review by oliver-sanders — APPROVED** [review]

**Comment by MetRonnie** [comment]
Rebasing to fix conflicts with master

**Comment by oliver-sanders** [comment]
Hmm, has Travis turned off? The tests haven't run.

**Comment by MetRonnie** [comment]
In https://travis-ci.org/github/metomi/isodatetime/requests it says

> GitHub payload is missing a merge commit (mergeable_state: "dirty", merged: false)

Not sure what that means but I will try rejiggering the PR

**Comment by oliver-sanders** [comment]
That did it.

**Pull request merged**